src/services/use-logger.ts

```diff
@@ -0,0 +1,42 @@
+import type { Guild, GuildMember, Role } from 'discord.js';
+
+import { useGuildsRepository } from '../db/use-guilds-repository.js';
+import { Logger } from './logger.js';
+
+const loggerMap = new Map<string, Logger>();
+
+export async function useLogger(guild: Guild): Promise<Logger> {
+  const logger = loggerMap.get(guild.id);
+
+  if (logger) {
+    return logger;
+  }
+
+  const guildsRepository = useGuildsRepository();
+  const dbGuild = await guildsRepository.findById(guild.id);
+
+  if (!dbGuild?.logChannelId) {
+    throw new Error();
+  }
+
+  const channel = await guild.client.channels.fetch(dbGuild.logChannelId);
+
+  if (!channel?.isTextBased()) {
+    throw new Error();
```

Comment:
Errors need to have text and be more specific

Reply:
PLEASE